File: src/main/java/org/example/mod/ExampleMod.java
package org.example.mod;

import felis.kittens.core.CommonEntrypoint;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ExampleMod implements CommonEntrypoint {
    public static final Logger LOGGER = LoggerFactory.getLogger(ExampleMod.class);

    @Override
    public void onInit() {
        LOGGER.info("Example mod has been initialized");
    }
}

File: src/main/kotlin/org/example/mod/ExampleMod.kt
package org.example.mod

import felis.kittens.core.CommonEntrypoint
import felis.kittens.core.event.GameEvents
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object ExampleMod : CommonEntrypoint {
    val logger: Logger = LoggerFactory.getLogger(ExampleMod::class.java)

    override fun onInit() {
        GameEvents.Block.broken.register { ctx ->
            this.logger.info("Thou shalt not break block")
            ctx.isCancelled = true
        }
        this.logger.info("Example mod has been initialized")
    }
}
